### PR TITLE
upgrade to fastify 4.x

### PR DESCRIPTION
**Pull request by Froskk**
This works on my latest project on windows but haven't bothered to setup tests/commit hooks as I don't have the time to fix the setup

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -113,22 +114,27 @@ export default fastifyPlugin<FastifyAutoroutesOptions>(
     }
 
     if (!fs.existsSync(dirPath)) {
-      return next(new Error(`${ERROR_LABEL} dir ${dirPath} does not exists`))
```
Comment (Froskk) [inline src/index.ts:116]: Chose abitrary error code of `'1'` and `'2'`

File: src/index.ts [hunk 2]
```diff
@@ -113,22 +114,27 @@ export default fastifyPlugin<FastifyAutoroutesOptions>(
...
       )
+      return next(new CustomError())
     }
 
-    let routes = await glob(`${dirPath}/**/[!._]*/[!._]*.{ts,js}`)
-    const routesModules: Record<string, StrictResource> = {}
-
     // glob returns ../../, but windows returns ..\..\
-    routes = routes.map((route) => path.normalize(route).replace(/\\/g, '/'))
     dirPath = path.normalize(dirPath).replace(/\\/g, '/')
```
Comment (Froskk) [inline src/index.ts:133]: Replacing the path seperators here resolves the issue and glob returns the correct paths

File: src/index.ts [hunk 3]
```diff
@@ -113,22 +114,27 @@ export default fastifyPlugin<FastifyAutoroutesOptions>(
     }
 
     if (!fs.existsSync(dirPath)) {
-      return next(new Error(`${ERROR_LABEL} dir ${dirPath} does not exists`))
+      const CustomError = createError(
```
Comment (Froskk) [inline src/index.ts:117]: Chose abitrary error code of `'1'` and `'2'`

**Comment by GiovanniCardamone** [comment]
Sorry i have seen this a week ago but I am very busy with my life rn.

I'll merge and check this weekend, I'll create a release very soon.

Sorry again and thanks for the PR

**Comment by GiovanniCardamone** [comment]
i think @fastify/error should be installed as `dependencies` instead of `devDependencies` or whoever doesn't have it installed will fails

**Comment by GiovanniCardamone** [comment]
merged, released as `fastify-autorotes@3.0.0`

sorry for delay

**Pull request merged**